File: sample/src/main/java/vn/tiki/sample/util/Assets.java
package vn.tiki.sample.util;

import android.content.Context;
import io.reactivex.Single;
import java.io.InputStream;
import okio.BufferedSource;
import okio.Okio;

public final class Assets {

  private final Context context;

  public Assets(Context context) {
    this.context = context;
  }

  public Single<BufferedSource> read(String file) {
    return Single.fromCallable(() -> {
      final InputStream inputStream = context.getAssets()
          .open(file);
      return Okio.buffer(Okio.source(inputStream));
    });
  }

}

File: sample/src/main/java/vn/tiki/sample/util/Assets.kt
package vn.tiki.sample.util

import android.content.Context
import io.reactivex.Single
import okio.BufferedSource
import okio.Okio

class Assets(private val context: Context) {

  fun read(file: String): Single<BufferedSource> {
    return Single.fromCallable {
      val inputStream = context.assets.open(file)
      Okio.buffer(Okio.source(inputStream))
    }
  }

}
